hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/MarkerBasedRollbackStrategy.java

```diff
@@ -75,64 +87,150 @@ public List<HoodieRollbackRequest> getRollbackRequests(HoodieInstant instantToRo
       List<String> markerPaths = MarkerBasedRollbackUtils.getAllMarkerPaths(
           table, context, instantToRollback.getTimestamp(), config.getRollbackParallelism());
       int parallelism = Math.max(Math.min(markerPaths.size(), config.getRollbackParallelism()), 1);
-      return context.map(markerPaths, markerFilePath -> {
+      List<HoodieRollbackRequest> rollbackRequests = context.map(markerPaths, markerFilePath -> {
         String typeStr = markerFilePath.substring(markerFilePath.lastIndexOf(".") + 1);
         IOType type = IOType.valueOf(typeStr);
+        String fileNameWithPartitionToRollback = WriteMarkers.stripMarkerSuffix(markerFilePath);
+        Path fullFilePathToRollback = new Path(basePath, fileNameWithPartitionToRollback);
+        String partitionPath = FSUtils.getRelativePartitionPath(new Path(basePath), fullFilePathToRollback.getParent());
         switch (type) {
           case MERGE:
           case CREATE:
-            String fileToDelete = WriteMarkers.stripMarkerSuffix(markerFilePath);
-            Path fullDeletePath = new Path(basePath, fileToDelete);
-            String partitionPath = FSUtils.getRelativePartitionPath(new Path(basePath), fullDeletePath.getParent());
-            return new HoodieRollbackRequest(partitionPath, EMPTY_STRING, EMPTY_STRING,
-                Collections.singletonList(fullDeletePath.toString()),
+            String fileId = null;
+            String baseInstantTime = null;
+            if (FSUtils.isBaseFile(fullFilePathToRollback)) {
+              HoodieBaseFile baseFileToDelete = new HoodieBaseFile(fullFilePathToRollback.toString());
+              fileId = baseFileToDelete.getFileId();
+              baseInstantTime = baseFileToDelete.getCommitTime();
+            } else if (FSUtils.isLogFile(fullFilePathToRollback)) {
+              // TODO: HUDI-1517 may distinguish log file created from log file being appended in the future @guanziyue
+              // Now it should not have create type
+              checkArgument(type != IOType.CREATE, "Log file should not support create io type now");
+              checkArgument(type != IOType.MERGE, "Log file should not support merge io type");
```

Comment:
i'm not sure i follow the validation here.. these validations are under case MERGE and CREATE. One of them will fail right?